### ISSUE TITLE
Subscription ActionCable channel incorrect when applying scopes?

### ISSUE THREAD

**Issue by markymc** [rmosolgo/graphql-ruby#3116]
@rmosolgo as requested via https://graphql.slack.com/archives/C07MVL1AB/p1597804676042400, I'm creating an issue here.

**Describe the bug**

Hi all, I’m having some trouble with subscriptions using ActionCable, ActionCableLink, and Apollo when attempting to apply scopes.
When I create a simple subscription, such as:

```
# frozen_string_literal: true
class Subscriptions::NewOrder < Subscriptions::BaseSubscription
  field :order, Types::OrderType, null: false
end
```

with my subscription type as so:

```
# frozen_string_literal: true
class Types::SubscriptionType < Types::BaseObject
  extend GraphQL::Subscriptions::SubscriptionRoot
  field :new_order,
        subscription: Subscriptions::NewOrder
end
```

and subscribe to it using:

```
const NEW_ORDER_SUBSCRIPTION = gql`
  subscription NewOrder {
    newOrder {
      order {
        ...Order
      }
    }
  }
  ${OrderSidebarItem.fragments.order}
`;
...
getOrdersQuery.subscribeToMore({
    document: NEW_ORDER_SUBSCRIPTION,
    updateQuery: (prev, { subscriptionData }) => {
      const order = subscriptionData.data?.newOrder.order;
      console.log({ order, prev });
      ...do_stuff_here
    },
  });
```

and trigger using:

```
    Schema.subscriptions.trigger(
      :new_order,
      {},
      { order: self }
    )
```

I get event triggering no problem. `ActionCable is broadcasting to: [ActionCable] Broadcasting to graphql-event::newOrder:: "{\"order\":{\"__gid__\":\"Z2lkOi8vZm9vZC1lL09yZGVyLzIyMw\"},\"__sym_keys__\":[\"order\"]}"`

However, I want to scope them to a particular user, therefore:

```
# frozen_string_literal: true
class Subscriptions::NewOrder < Subscriptions::BaseSubscription
  field :order, Types::OrderType, null: false
  subscription_scope :current_user
end
```

and

```
    Schema.subscriptions.trigger(
      :new_order,
      {},
      { order: self },
      scope: user
    )
```
The problem seems to be that ActionCable is then broadcasting to a different channel now:
`[ActionCable] Broadcasting to graphql-event:Z2lkOi8vZm9vZC1lL1Jlc3RhdXJhbnQvMQ:newOrder:: "{\"order\":{\"__gid__\":\"Z2lkOi8vZm9vZC1lL09yZGVyLzIyMg\"},\"__sym_keys__\":[\"order\"]}"`

But I don’t think my client (using ActionCableLink with Apollo 3) is subscribed to that channel (Redis shows `1597804631.180828 [1 [::1]:63465] "subscribe" "graphql-event::newOrder:"`.

Am I missing something important here? Is this a client-side problem or server-side? Thanks so much for any help you can give! :bow:

**Versions**

`graphql` version: 1.10.0
`rails` (or other framework): 6.0.3.2
other applicable versions (`graphql-batch`, etc)

**GraphQL schema**

See above

**GraphQL query**

See above

**Steps to reproduce**

Subscribe to the subscription and create a new Order (which triggers).

**Expected behavior**

The user would receive an update via the subscription.

**Actual behavior**

The user doesn't get an update, I think because the ActionCable channel is incorrect on one side.

</details>

**Additional context**

None

**Comment by rmosolgo** [comment]
Thanks for sharing those details. You can see how the event topic (which is used as a channel name) includes the `scope` at the beginning: 



It looks like the `.trigger` call has the proper scope: 

```
[ActionCable] Broadcasting to graphql-event:Z2lkOi8vZm9vZC1lL1Jlc3RhdXJhbnQvMQ:newOrder:: "{\"order\":{\"__gid__\":\"Z2lkOi8vZm9vZC1lL09yZGVyLzIyMg\"},\"__sym_keys__\":[\"order\"]}" 
                                            ^^ scope  (user global id)      ^^
```

But the `subscribe` call doesn't have it: 

```
1597804631.180828 [1 [::1]:63465] "subscribe" "graphql-event::newOrder:"
                                                            ^^ blank string instead of scope
```

That suggests to me that, when subscribing, `context[:current_user]` is nil. Can you confirm that `context[:current_user]` is the same object as the `scope: user` object? For example:

```diff
  # frozen_string_literal: true
  class Subscriptions::NewOrder < Subscriptions::BaseSubscription
    field :order, Types::OrderType, null: false
+   def subscribe
+     pp context[:current_user]
+     super 
+   end 
  end
```

and 

```diff 
+   pp user 
    Schema.subscriptions.trigger(
      :new_order,
      {},
      { order: self },
      scope: user
    )
```

What do you see when you look into that?

**Comment by markymc** [comment]
@rmosolgo Thanks very much for your feedback. You're completely right, I hadn't realised that I needed to setup an ActionCable::Connection::Base to handle auth as it wouldn't use the standard auth we had setup via the GQL endpoint. I ended up using [https://stackoverflow.com/questions/35501931/send-auth-token-for-authentication-to-actioncable/36534210#36534210](this technique) on client and server and it seems to be working now. I'm sorry for taking your time, and appreciate you pointing me in the right direction.
